pkg/process/runner/runner.go

```diff
@@ -393,6 +398,53 @@ func (l *CheckRunner) basicRunner(c checks.Check) func() {
 	}
 }
 
+// getConnectionsCheckInterval validates 'process_config.connections_capacity_check_interval' (the "capacity interval").
+// It must be between 10s-30s (inclusive) and be a factor of defaultInterval (the main connections check interval).
+// If both are true, the capacity interval is returned.
+// If explicitly configured by the user but fails validation, a specific warning is logged and defaultInterval is returned.
+// Otherwise (e.g. not configured, or derived value is invalid), defaultInterval is returned.
+func (l *CheckRunner) getConnectionsCheckInterval(checkName string, defaultInterval time.Duration) time.Duration {
+	const minAllowedInterval = 10 * time.Second
+	const maxAllowedInterval = 30 * time.Second
```

Comment:
the max must match the configuration value of connection check.
We can set the connections check to report every 1m if we want, but the current code ignore that and forces us to use 30s or lower).

Also, the minimum interval can cause a violation as well, as if we set a connection check interval to 10s (we have that in a single customer due to high load), then we have 2 issues -
1. Race between the new endpoint and the connections check endpoint (cc: see my comment in the RFC about reusing the same endpoint)
2. you hard-coded prevent the connection check to be less than 10s